src/core/io/eIO/EConfig.h

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Maximilian Scharf
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Maximilian Scharf - Initial implementation.
+ *******************************************************************************/
+
+#pragma once
+
+#include <forward_list>
+#include "extevhandlerhelper.h"
+#include "ESpec.h"
+
+#include "devlog.h"
+
+class FORTE_eGenAdapter;
+/**
+ * This class handles the registration of event-triggers on the controller
+ * and maintains a record of all registered event-triggers from this instance, within this instance.
+ */
+class EConfig {
+  private:
+    CProcessInterfaceFB *peer;
+
+  protected:
+    std::forward_list<ESpecBase*> eventGenList;
```

Comment:
a forwardlist seems a bit an overkill here. even if the list is changing once in a while a vector would be more efficient and performant.